qualtran/bloqs/phase_estimation/lp_resource_state.py

```diff
@@ -0,0 +1,179 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Resource states proposed by A. Luis and J. Peřina (1996) for optimal phase measurements"""
+from functools import cached_property
+from typing import Set, TYPE_CHECKING
+
+import attrs
+import cirq
+import numpy as np
+from numpy._typing import NDArray
+
+from qualtran import bloq_example, BloqDocSpec, GateWithRegisters, QUInt, Register, Side, Signature
+from qualtran.bloqs.basic_gates import CZPowGate, Hadamard, OnEach, Ry, Rz, XGate, ZPowGate
+from qualtran.bloqs.mcmt import MultiControlPauli
+from qualtran.cirq_interop import CirqGateAsBloq
+from qualtran.cirq_interop.t_complexity_protocol import TComplexity
+
+if TYPE_CHECKING:
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@attrs.frozen
+class _LPRSHelper(GateWithRegisters):
```

Comment:
Is it worthwhile making this public? It could be nice in a notebook to show this reproduces Eq. 19 for example?

Reply:
Done